platform/mbed_debug.h

```diff
@@ -18,49 +18,49 @@
  */
 #ifndef MBED_DEBUG_H
 #define MBED_DEBUG_H
-#include "device.h"
+#if DEVICE_STDIO_MESSAGES
+#include <stdio.h>
+#include <stdarg.h>
+#endif
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#if DEVICE_STDIO_MESSAGES
-#include <stdio.h>
-#include <stdarg.h>
 
 /** Output a debug message
  *
  * @param format printf-style format string, followed by variables
  */
 static inline void debug(const char *format, ...) {
+#if DEVICE_STDIO_MESSAGES && !defined(NDEBUG)
```

Comment:
NDEBUG is not defined in the default profile, not sure how that affects mbed 2
https://github.com/ARMmbed/mbed-os/blob/master/tools/profiles/develop.json